### PR TITLE
feat: PragmaGix monitoring

### PR DESCRIPTION
**Pull request by azurwastaken**
Resolves: #41

## Work Done
* Added new metrics for the Dispatched events,
* Supports new Pragma Devnet tables,

### CODE REVIEW

File: src/models.rs [hunk 1]
```diff
@@ -90,32 +90,45 @@ pub struct VrfRequest {
...
     pub updated_at: NaiveDateTime,
     pub updated_at_tx: String,
-}
\ No newline at end of file
+}
+
+#[derive(Debug, Queryable, Selectable, QueryableByName)]
+#[diesel(table_name = crate::schema::pragma_devnet_dispatch_event)]
+#[diesel(check_for_backend(diesel::pg::Pg))]
+pub struct PragmaDevnetDispatchEvent {
```
Comment (EvolveArt) [inline src/models.rs:125]:
```suggestion
pub struct FeedDispatch {
```

File: src/models.rs [hunk 2]
```diff
@@ -90,32 +90,45 @@ pub struct VrfRequest {
...
+    pub network: String,
+    pub block_hash: String,
+    pub block_number: i64,
+    pub block_timestamp: NaiveDateTime,
+    pub transaction_hash: String,
+    pub hyperlane_message_nonce: BigDecimal,
+    pub feeds_updated: Option<Vec<String>>,
+    pub _cursor: (Bound<i64>, Bound<i64>),
```
Comment (EvolveArt) [inline src/models.rs:133]: cursor shouldn't be a range in that case no?

File: src/processing/spot.rs [hunk 3]
```diff
@@ -52,6 +53,14 @@ pub async fn process_data_by_pair(
                 .first(&mut conn)
                 .await?
         }
+        NetworkName::PragmaDevnet => {
+            testnet_dsl::spot_entry
+                .filter(testnet_dsl::network.eq("pragma-devnet"))
+                .filter(testnet_dsl::pair_id.eq(pair.clone()))
+                .order(testnet_dsl::block_timestamp.desc())
```
Comment (EvolveArt) [inline src/processing/spot.rs:60]: testnet_dsl ?

File: src/processing/spot.rs [hunk 4]
```diff
@@ -166,6 +185,15 @@ pub async fn process_data_by_publisher(
...
         NetworkName::Testnet => {
             testnet_dsl::spot_entry
+                .filter(testnet_dsl::network.eq("testnet"))
+                .filter(testnet_dsl::publisher.eq(publisher.clone()))
+                .order(testnet_dsl::block_timestamp.desc())
+                .first(&mut conn)
+                .await?
+        }
+        NetworkName::PragmaDevnet => {
+            testnet_dsl::spot_entry
```
Comment (EvolveArt) [inline src/processing/spot.rs:195]: you should use pragma_devnet_dsl I think

**Review by EvolveArt — CHANGES_REQUESTED** [review]

File: src/processing/spot.rs (hunk 4 above)
Comment (akhercha) [reply src/processing/spot.rs:195]: yeah sry, done

File: src/processing/spot.rs (hunk 3 above)
Comment (akhercha) [reply src/processing/spot.rs]: woaaa missed all of those, didn't spaw that we updated them. Corrected!

File: src/models.rs (hunk 2 above)
Comment (akhercha) [reply src/models.rs]: ah yeah u're right - updated

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**